libraries/provider_firewall_windows.rb

```diff
@@ -68,11 +68,29 @@ def whyrun_supported?
       windows_file.content build_rule_file(new_resource.rules['windows'])
       windows_file.run_action(:create)
 
-      # if the file was changed, restart iptables
-      if windows_file.updated_by_last_action?
+      # if the file was changed
+      # OR
+      # if current_rule_count is not equal to the number of rules in file
```

Comment:
As I tried to explain in the comments in the code, I am not implicitly or explicitly counting on the lines to check if default rules are applied or not. What I am trying to do is to check if the rules tampered out of chef. 

So if num of rules defined in chef is not equal to num of existing rules OR if these numbers are equal but default_rules property is enabled in provider, then we need to restart.. 

Only checking if file is updated or not does not provide a secure environment: Somebody may add / delete some rules manually and chef will not update to the desired state, as rules defined in chef and the rules in that file will be the same and file will not be updated.

I don't see any point that will break. What is your samples on "very prone to breaking" ?